google-cloud-clients/google-cloud-firestore/src/test/java/com/google/cloud/firestore/LocalFirestoreHelper.java

```diff
@@ -121,6 +126,52 @@
 
   public static final Precondition UPDATE_PRECONDITION;
 
+  public static class CompositeField {
+
+    public String stringField;
+    public PrimitiveField primitiveField;
+
+    public CompositeField(String stringField, PrimitiveField primitiveField) {
+      this.stringField = stringField;
+      this.primitiveField = primitiveField;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
`equals` method should have corresponding `hashCode`

Reply:
I wonder if we even need this equals method?